internal/mode/static/state/resolver/resolver.go

```diff
@@ -39,10 +40,15 @@ func NewServiceResolverImpl(client client.Client) *ServiceResolverImpl {
 	return &ServiceResolverImpl{client: client}
 }
 
-// Resolve resolves a Service and Port to a list of Endpoints.
+// Resolve resolves a Service and BackendRef Port to a list of Endpoints.
 // Returns an error if the Service or Port cannot be resolved.
-func (e *ServiceResolverImpl) Resolve(ctx context.Context, svc *v1.Service, port int32) ([]Endpoint, error) {
-	if svc == nil {
+//
+// svcNsName is guaranteed to be a valid NamespacedName from when it is called in configuration.go.
+// svcPort is guaranteed to be a valid non-empty ServicePort from when it is called in configuration.go
+func (e *ServiceResolverImpl) Resolve(ctx context.Context,
```

Comment:
nit: place each argument on its own line:
```suggestion
func (e *ServiceResolverImpl) Resolve(
    ctx context.Context,
    svcNsName types.NamespacedName,
    svcPort v1.ServicePort,
)
```

I wouldn't apply the suggestion directly through GitHub because I don't think it will work.